lib/vSphere/util/vm_helpers.rb

```diff
@@ -33,6 +33,127 @@ def powered_off?(vm)
         def suspended?(vm)
           get_vm_state(vm).eql?(VmState::SUSPENDED)
         end
+
+        # Enumerate VM snapshot tree
+        #
+        # This method returns an enumerator that performs a depth-first walk
+        # of the VM snapshot grap and yields each VirtualMachineSnapshotTree
+        # node.
+        #
+        # @param vm [RbVmomi::VIM::VirtualMachine]
+        #
+        # @return [Enumerator<RbVmomi::VIM::VirtualMachineSnapshotTree>]
+        def enumerate_snapshots(vm)
+          snapshot_info = vm.snapshot
+
+          if snapshot_info.nil?
+            snapshot_root = []
+          else
+            snapshot_root = snapshot_info.rootSnapshotList
+          end
+
+          recursor = lambda do |snapshot_list|
+            Enumerator.new do |yielder|
+              snapshot_list.each do |s|
+                # Yield the current VirtualMachineSnapshotTree object
+                yielder.yield s
+
+                # Recurse into child VirtualMachineSnapshotTree objects
+                children = recursor.call(s.childSnapshotList)
+                loop do
+                  yielder.yield children.next
+                end
+              end
+            end
+          end
+
+          recursor.call(snapshot_root)
+        end
+
+        # Create a named snapshot on a given VM
+        #
+        # This method creates a named snapshot on the given VM. This method
+        # blocks until the snapshot creation task is complete. An optional
+        # block can be passed which is used to report progress.
+        #
+        # @param vm [RbVmomi::VIM::VirtualMachine]
+        # @param name [String]
+        # @yield [Integer] Percentage complete as an integer. Called multiple
+        #   times.
+        #
+        # @return [void]
+        def create_snapshot(vm, name)
+          task = vm.CreateSnapshot_Task(
+            name: name,
+            memory: false,
+            quiesce: false)
```

Comment:
I picked these settings because they result in the fastest snapshots and closely mirror the "disk-only" behavior of other cloud platforms such as AWS and OpenStack. However, I'm open to using different settings.

One thing to note is that the `vagrant snapshot save` and `vagrant snapshot push` commands don't provide any option handling, so there's no way to adjust these settings from the CLI.